### PR TITLE
feat(core): entity temporary components

### PR DESCRIPTION
**Pull request by Miou-zora**
I added a way to create component that should have a short lifecycle (1 cycle of run systems) that can allow to update entities only if needed.

For exemple, for buttons, I don't want to check all entities if there state has changed in every system that use button component so went I hover or click on a button, I've added a temporary component `HasChanged<Button>` to only check these entities.

To add this feature inside your project, just add `Entity::RemoveTemporaryComponents` at the end of all of your systems and call `AddTemporaryComponent` to add a temporary component.

### CODE REVIEW

**Review by ripel2 — CHANGES_REQUESTED** [review]
great idea but it could be more clear to rename those temporary components as "tags" everywhere for clarity

File: src/engine/entity/Entity.hpp [hunk 1]
```diff
@@ -68,6 +69,58 @@ class Entity {
...
+    /**
+     * Utility method to add a temporary component to an entity.
+     * Temporary component are removed when calling RemoveTags system.
+     *
+     * @tparam  TTempComponent  type to add to registry
+     * @param   registry        registry used to store the component
+     * @return  reference of the added component
+     * @see     RemoveTags
+     */
+    template <typename TTempComponent> inline decltype(auto) AddTemporaryComponent(Registry &registry)
```
Comment (ripel2) [inline src/engine/entity/Entity.hpp:81]: here use something like `template <typename TagComponent>`

File: src/engine/entity/Entity.hpp [hunk 2]
```diff
@@ -96,5 +149,6 @@ class Entity {
 
   private:
     entity_id_type _entity;
+    inline static std::unordered_map<std::type_index, std::function<void(Registry &)>> temporaryComponent = {};
```
Comment (ripel2) [inline src/engine/entity/Entity.hpp:152]: name the map tagComponents as well

Comment (Miou-zora) [reply src/engine/entity/Entity.hpp:152]: It's not a tag, I renamed `RemoveTags` to `RemoveTemporaryComponents`.

**Review by ripel2 — APPROVED** [review]

**Pull request merged**